packages/client/lib/functions/placeShip.tsx

```diff
@@ -29,67 +31,53 @@ const placeShip = (
       console.log("goesOutsideGrid");
       return true;
     }
-    return false;
+    return undefined;
   };
 
   /** True if neighbouring x or y + ... ship length coordinates is not occupied + !isOutsideGrid */
   const isLocationPlaceable = (x: number, y: number) => {
-    if (goesOutsideGrid()) return false;
-
     if (orientation === "horizontal") {
       for (let i = x; i < xrange; i++) {
-        if (board[i][y] !== 0) {
-          return false;
+        console.log("Checking for ship at " + i + ", " + y);
+        const foundShips = shipsInCoord(shipInfo, [i, y]);
```

Comment:
Instead of comparing the location with board's value, this checks whether the location is already being used in a player's ship.

Reply:
Yeah, since we don't have any explicit "board values" that we are actually storing, so the checking makes sense.